bls/globals.go

```diff
@@ -150,3 +150,15 @@ func EvaluatePolyInEvaluationForm(yFr *Fr, poly []Fr, x *Fr, rootsOfUnity []Fr,
 	// (x**WIDTH - 1) / WIDTH  *  sum_(i=0)^WIDTH  (f(DOMAIN[i]) * DOMAIN[i]) / (x - DOMAIN[i])
 	MulModFr(yFr, &y, &tmp)
 }
+
+func PolyLinComb(vectors [][]Fr, scalars []Fr) []Fr {
+	r := make([]Fr, len(vectors[0]))
+	for i := 0; i < len(vectors); i++ {
+		var tmp Fr
+		for j := 0; j < len(r); j++ {
+			MulModFr(&tmp, &vectors[i][j], &scalars[i])
+			AddModFr(&r[j], &r[j], &tmp)
+		}
+	}
```

Comment:
```suggestion
	for _, v := range vectors {
		var tmp Fr
		for i := 0; i < len(r); i++ {
			MulModFr(&tmp, &v[i], &scalars[i])
			AddModFr(&r[i], &r[i], &tmp)
		}
	}
```

Iterating over the vectors and copying the slice header `v` (24 bytes, not the full thing) of each vector is probably fine in terms of performance, and makes it read more like the spec. And maybe even reduces the lookup overhead in the inner loop (not sure what compiler would do).